akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -417,6 +417,25 @@ trait FlowOps[+Out, +Mat] {
   def recover[T >: Out](pf: PartialFunction[Throwable, T]): Repr[T] = andThen(Recover(pf))
 
   /**
+    * RecoverWith allows to switch to alternative Source on flow failure. It can do switch
+    * several times by materializing each time Source that `pf` returns.
+    * Since the underlying failure signal onError arrives out-of-band, it might jump over existing elements.
+    * This stage can recover the failure signal, but not the skipped elements, which will be dropped.
+    *
+    * '''Emits when''' element is available from the upstream or upstream is failed and element is available
+    * from alternative Source
+    *
+    * '''Backpressures when''' downstream backpressures
+    *
+    * '''Completes when''' upstream completes or upstream failed with exception pf can handle
+    *
+    * '''Cancels when''' downstream cancels
+    *
+    */
+  def recoverWith[T >: Out](pf: PartialFunction[Throwable, Graph[SourceShape[T], Mat @uncheckedVariance]]): Repr[T] =
```

Comment:
I will change this o NotUsed as we don't use materialization value here. 
Regarding ```ignoreMaterializedValue```, don't think we need it for this case as it's only casting to NotUsed for materialization